components/hab/src/license.rs

```diff
@@ -0,0 +1,163 @@
+    pub accepting_product_version: String,
+    pub user: Option<String>,
+    pub file_format: String,
+}
+
+impl LicenseData {
+    pub fn new() -> Self {
+        LicenseData { date_accepted: Utc::now(),
+                      accepting_product: String::from("hab"),
+                      accepting_product_version: super::VERSION.to_string(),
+                      user: get_current_username(),
+                      file_format: String::from(LICENSE_FILE_FORMAT_VERSION), }
+    }
+}
+
+pub fn check_for_license_acceptance_and_prompt(ui: &mut UI) -> Result<()> {
+    if license_exists() || env_var_present()? {
+        return Ok(());
+    }
+
+    ui.heading("+---------------------------------------------+")?;
+    ui.heading("            Chef License Acceptance")?;
+    ui.br()?;
+    ui.info("Before you can continue, 1 product license must be accepted.")?;
+    ui.info("View the license at https://www.chef.io/end-user-license-agreement")?;
+    ui.br()?;
+    ui.info("License that needs accepting:")?;
+    ui.br()?;
+    ui.info("  * Habitat")?;
+    ui.br()?;
+
+    if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+        accept_license(ui)
+    } else {
+        ui.br()?;
+        ui.info("If you do not accept this license you will not be able to use Chef products.")?;
+        ui.br()?;
+
+        if ui.prompt_yes_no("Do you accept the 1 product license?", Some(false))? {
+            accept_license(ui)
+        } else {
+            ui.br()?;
+            ui.heading("+---------------------------------------------+")?;
+            Err(Error::LicenseNotAccepted)
+        }
+    }
+}
+
+pub fn accept_license(ui: &mut UI) -> Result<()> {
+    if license_exists() {
+        ui.info("You have already accepted the license.")?;
+        return Ok(());
+    }
+
+    ui.br()?;
+    ui.info("Accepting 1 product license...")?;
+
+    write_license_file()?;
+
+    ui.status(ui::Status::Custom(ui::Glyph::CheckMark, String::from("")),
+              "1 product license accepted.")?;
+    ui.br()?;
+    ui.heading("+---------------------------------------------+")?;
+
+    Ok(())
+}
+
+fn license_path() -> PathBuf {
+    let hab_dir = if am_i_root() {
+        PathBuf::from(&*FS_ROOT_PATH).join("hab")
+    } else if let Some(home) = dirs::home_dir() {
+        home.join(".hab")
+    } else {
+        panic!("No home directory available. Unable to find a suitable place to write a license \
+                file.");
+    };
+
+    hab_dir.join("accepted-licenses")
+}
+
+fn env_var_present() -> Result<bool> {
+    match env::var(LICENSE_ACCEPT_ENVVAR) {
+        Ok(val) => {
+            if &val == "accept" {
+                write_license_file()?;
+                Ok(true)
+            } else if &val == "accept-no-persist" {
+                Ok(true)
+            } else {
+                Ok(false)
+            }
+        }
+        Err(_) => Ok(false),
+    }
+}
+
+fn write_license_file() -> Result<()> {
```

Comment:
That's my thought.  If they have accepted the license in a way that necessitates the writing of the file (instead of e.g. `export HAB_LICENSE=accept-no-persist`), then I think that makes sense, no?

Reply:
yeah, makes sense